src/pipelinedefinition.cpp

```diff
@@ -387,8 +387,8 @@ class NodeValidator {
                     gather);
                 return StatusCode::PIPELINE_NODE_GATHER_FROM_NOT_DEMULTIPLEXER;
             }
-            if (it->kind == NodeKind::ENTRY) {
-                SPDLOG_LOGGER_ERROR(dag_executor_logger, "Validation of pipeline: {} definition failed. Node name: {}, have gather_from: {}. Gathering from entry node is not implemented yet",
+            if (it->kind == NodeKind::ENTRY && dependantNodeInfo.kind != NodeKind::EXIT) {
+                SPDLOG_LOGGER_ERROR(dag_executor_logger, "Validation of pipeline: {} definition failed. Node name: {}, have gather_from: {}. Gathering from entry node is not allowed for non exit node",
```

Comment:
It might be useful to gather from entry node manually. Just an idea.

Reply:
Will add test and potentially enable it in separate MR